### PR TITLE
DP-1770 DP-1771

### PR DESCRIPTION
**Pull request by kowallen**
DP-1770: Fix not committing offset for heterogeneous topics
DP-1771: Enable flattener by default, fix parsing for Bad Requests response

Done:
[x] Manual tests for heterogeneous topics

### CODE REVIEW

File: connector/src/main/scala/com/celonis/kafka/connect/ems/storage/EmsUploader.scala [hunk 1]
```diff
@@ -160,7 +160,7 @@ class EmsUploader[F[_]](
     )
     A.delay(
       logger.error(
-        s"Failed to upload the file:$file. Status code:${response.status.show}, Error:${error.msg}",
+        s"Failed to upload the file:$file. Status code:${response.status.show}, Error:${error.msg}. Error: ${throwable.getMessage}",
```
Comment (afiore) [inline connector/src/main/scala/com/celonis/kafka/connect/ems/storage/EmsUploader.scala:163]: Does including `error` in this log line add much information?

```suggestion
        s"Failed to upload the file:$file. Status code:${response.status.show}, Exception: ${throwable.getMessage}",
```

Note: I would also do the same logging we have above, in `genericError(...)`

File: connector/src/main/scala/com/celonis/kafka/connect/ems/storage/WriterBuilder.scala [hunk 2]
```diff
@@ -83,17 +94,49 @@ class WriterBuilderImpl(
...
+      state,
+    )
+  }
+
+  /** Creates a new [[Writer]] whenever the schema rollback happened. It also means that the previous file was sent, so
+    * committedOffset is set lastOffset as a from previous [[Writer]]
+    *
+    * @param record
+    *   \- An instance of [[Record]]
+    * @param writer
```
Comment (afiore) [inline connector/src/main/scala/com/celonis/kafka/connect/ems/storage/WriterBuilder.scala:122]: Does this argument need to be the whole writer? I guess supplying a `Offset` would be just enough (as per [principle of least knowledge](https://en.wikipedia.org/wiki/Law_of_Demeter))

File: connector/src/test/scala/com/celonis/kafka/connect/ems/storage/EmsUploaderTests.scala [hunk 3]
```diff
@@ -167,7 +154,63 @@ class EmsUploaderTests extends AnyFunSuite with Matchers {
...
         }
-    }
+    }.unsafeRunSync()
+  }
+
+  test("handles non json responses") {
+    val port             = 21212
+    val auth             = "this is auth"
+    val targetTable      = "tableA"
+    val path             = "/api/push/br"
```
Comment (afiore) [inline connector/src/test/scala/com/celonis/kafka/connect/ems/storage/EmsUploaderTests.scala:164]: minor: let's rename this path even more explicitly e.g. `respond-with-bad-request`

**Review by afiore — CHANGES_REQUESTED** [review]

File: connector/src/main/scala/com/celonis/kafka/connect/ems/storage/WriterBuilder.scala (hunk 2 above)
Comment (kowallen) [reply connector/src/main/scala/com/celonis/kafka/connect/ems/storage/WriterBuilder.scala:122]: From one side it doesn't have to, but with Writer in signature you know that the new one must be created from already existing one, so i find it more readable.

**Pull request merged**